src/main/scala/mesosphere/marathon/api/akkahttp/v2/LeaderController.scala

```diff
@@ -0,0 +1,70 @@
+package mesosphere.marathon
+package api.akkahttp.v2
+
+import akka.http.scaladsl.server.Route
+import mesosphere.marathon.api.akkahttp.{ Controller, Rejections }
+import mesosphere.marathon.core.election.ElectionService
+import mesosphere.marathon.plugin.auth._
+import com.typesafe.scalalogging.StrictLogging
+import mesosphere.marathon.raml.RuntimeConfiguration
+import mesosphere.marathon.storage.repository.RuntimeConfigurationRepository
+
+import scala.async.Async._
+import scala.concurrent.ExecutionContext
+
+case class LeaderController(
+    val electionService: ElectionService,
+    val runtimeConfigRepo: RuntimeConfigurationRepository)(
+    implicit
+    val authenticator: Authenticator,
+    val authorizer: Authorizer,
+    val executionContext: ExecutionContext) extends Controller with StrictLogging {
+
+  import mesosphere.marathon.api.akkahttp.Directives._
+  import mesosphere.marathon.api.akkahttp.EntityMarshallers._
+
+  def leaderInfo(): Route =
+    authenticated.apply { implicit identity =>
+      authorized(ViewResource, AuthorizedResource.SystemConfig).apply {
+        electionService.leaderHostPort match {
+          case None => reject(Rejections.EntityNotFound.leader())
+          case Some(leader) => complete(raml.LeaderInfo(leader))
+        }
+      }
+    }
+
+  def deleteLeader(): Route =
+    authenticated.apply { implicit identity =>
+      authorized(UpdateResource, AuthorizedResource.SystemConfig).apply {
+        parameters('backup.?, 'restore.?) { (backup, restore) =>
+          // This follows the LeaderResource implementation. Seems like a bug to me. Why should we not proxy the request?
+          if (electionService.isLeader) {
+
+            //TODO: validate backup and restore parameters
```

Comment:
@meln1k, how should we do this validation?

Reply:
There are 2 options: https://doc.akka.io/docs/akka-http/current/scala/http/routing-dsl/directives/misc-directives/validate.html and https://doc.akka.io/docs/akka-http/current/scala/http/routing-dsl/case-class-extraction.html#case-class-validation

Reply:
I like the first one. Whatever you chose - by default akka http handles validation errors by throwing `ValidationRejection` and converting them to `HTTP 400 Bad Request` in default rejection handler. Marathon (at least for `v2/leader?backup=foo`) returns a `HTTP 422 Unprocessable Entity` with:
```
{
    "message": "Object is not valid",
    "details": [
        {
            "path": "/",
            "errors": [
                "Invalid URI or unsupported scheme: foo"
            ]
        }
    ]
}
```

Reply:
Hm, I thought we have validation with wix? Are we using the Akka HTTP validations for other models as well?

Personally I really do not like that we use that RAML model deep down in the storage layer.